preprocessing/preferences.py

```diff
@@ -0,0 +1,120 @@
+
+import random
+from preprocessing.xml_loader import *
+
+
+def generate_preferences(staff, time_set, num_weekly_preferences, preferences_dur):
+    """
+    Generates random preferences for each employee. The result is normalized to 1
+    Note: the result is identical between different runs, as using the same initial seed will
+    guarantee the same result
+    :param staff: data structure containing employees
+    :param time_set: data structure containing time_step, weeks and needed time_periods-sets
+    :param num_weekly_preferences: (min, max)-tuple representing bounds on the number of requests
+    per employee per week
+    :param preferences_dur: (min, max)-tuple representing bounds on the durations of the preference
+    :return: a dict with keys [e][t], representing employee and time respectively
+    """
+
+    # Ensuring persistent results between different runs. The seed can be any integer
+    random.seed(1)
+
+    employees = staff["employees"]
+    time_step = time_set["step"]
+    time_periods = time_set["periods"][0]
+    time_periods_in_day = time_set["periods"][2]
+    weeks = time_set["weeks"]
+
+    # Duration in hours is converted to duration in time_steps
+    preferences_dur = [int(preferences_dur[0]/time_step), int(preferences_dur[1]/time_step)]
+    preferences = initialize_preferences(employees, time_periods)
+
+    # Generating preferences
+    for employee in employees:
+        number_of_preferences = 0
+
+        for week in weeks:
+            used_time_periods = []
+            weekly_preferences = int(random.randint(num_weekly_preferences[0],
+                                                    num_weekly_preferences[1]))
+            realized_dur = get_realized_durations(preferences_dur, weekly_preferences)
+
+            for dur in realized_dur:
+
+                # Positive and negative values signifies preference for work and free,
+                # but unsure which value signifies what
```

Comment:
Positive values resemble a desire to work and negative values resemble a desire for off-time.